string name
IntentRequire[] requires
string[] at_least_one
IntentOptional[] optional
